### PR TITLE
Avoid double-firing the hard shutdown.

### PR DESCRIPTION
**Pull request by Lukasa**
See: https://twistedmatrix.com/trac/ticket/9157

### CODE REVIEW

File: src/twisted/web/test/test_http2.py [hunk 1]
```diff
@@ -2570,3 +2570,20 @@ def test_losingConnectionWithNoAbortTimeOut(self):
         reactor.advance(2**32)
         self.assertTrue(transport.disconnecting)
         self.assertFalse(transport.disconnected)
+
+    def test_connectionLostAfterForceClose(self):
```
Comment (hawkowl) [inline src/twisted/web/test/test_http2.py]: This needs two spaces between method definitions

**Review by hawkowl — APPROVED** [review]
Fix the nitpick then land.

**Comment by glyph** [comment]
🎉  thank you 🎉

**Comment by psi29a** [comment]
When is the next release of Twisted with this included? :)

**Comment by hawkowl** [comment]
Soon™ (worst case, next week at PyCon AU there will be a prerelease)

**Comment by psi29a** [comment]
Thank you! :)

**Pull request merged**